### ISSUE TITLE
Extra character `,` on end of useState() gives mediocre error message.

### ISSUE THREAD

**Issue by LeeWannacott** [quick-lint/quick-lint-js#639]
Doesn't tell me to remove extra character `,` on the end. 

![image](https://user-images.githubusercontent.com/49783296/153363872-7eabfeed-8a28-4035-86d0-cc658dc69b32.png)

This should give an error saying unexpected token and missing semicolon like below:
![image](https://user-images.githubusercontent.com/49783296/153366426-1323b06d-acb6-427d-94b0-acbb47198d07.png)

**Comment by LeeWannacott** [comment]
![image](https://user-images.githubusercontent.com/49783296/153366426-1323b06d-acb6-427d-94b0-acbb47198d07.png)
This should give an error saying unexpected token and missing semicolon.

**Comment by strager** [comment]
I think we can implement this using the following logic: if `const` is after `,` when parsing variable bindings (`parser::parse_and_visit_let_bindings` in `parse-statement-inl.h`), reporting a different error than E0024 (`error_let_with_no_bindings`). I think E0036 is similar.

**Comment by ooblegork** [comment]
Hi, strager. Since #707 is resolved, I would like to give this issue a shot.

I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by ooblegork** [comment]
I would like to give this issue a shot.

I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by ooblegork** [comment]
Please remove me from this. 

You called me retarded in your chat over a simple question and I refuse to deal with toxic individuals in any capacity.

**Comment by ooblegork** [comment]
Hey, remove me from this issue. You called me a retard in your stream chat despite no provocation, and I don't want to work with immature, toxic people. Thanks.

**Comment by Moshahrani** [comment]
> I think we can implement this using the following logic: if `const` is after `,` when parsing variable bindings (`parser::parse_and_visit_let_bindings` in `parse-statement-inl.h`), reporting a different error than E0024 (`error_let_with_no_bindings`). I think E0036 is similar.

Hi, is this task closed or still available??? Thanks

**Comment by strager** [comment]
@Moshahrani You can work on it if you want.

**Comment by Moshahrani** [comment]
> @Moshahrani You can work on it if you want.

Thanks!

**Comment by strager** [comment]
@Moshahrani Did you make progress? Do you need any help?

**Comment by matttheus** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by matttheus** [comment]
I know is not related to this issue, but I found a strange behavior while I was playing around.

I created try.jsx file with the following content:

```jsx
import React, { useState } from 'react'

export const App = () => {
  const [state, setState] = useState(),
  return (
    <div>hello</div>
  )
}
``` 

I run the command:
```bash
./build/quick-lint-js try/try.jsx
```

The error I got was:

```bash
try/try.jsx:4:3: error: const with no bindings [E0024]
```

This error is correct? I'm misunderstanding something?

**Comment by matttheus** [comment]
I'm still working on this issue

**Comment by matttheus** [comment]
Hey @strager I was not able to make progress on this issue, I will give up for now

**Comment by HicaroD** [comment]
Is this issue still opened by mistake?

I tried to simulate the bug in the QLJS playground:

![image](https://github.com/quick-lint/quick-lint-js/assets/75737377/0e094082-074c-443b-976f-602256ea9b96)

**Comment by strager** [comment]
The bug is still present. Here's a repro:
```javascript
function useState() {}

const [a, setA] = useState(),
const [b, setB] = useState();
```

**Comment by skdoosh** [comment]
I am assuming this is still open and unassigned, since the playground still shows the bug with the reproduction code.

I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
@skdoosh, sorry, I don't have the time & energy to review your patch. I can give the funds as compensation for your work; email me at s@strager.net
